src/Symfony/Component/HttpClient/HttpClient.php

```diff
@@ -31,6 +32,10 @@ final class HttpClient
      */
     public static function create(array $defaultOptions = [], LoggerInterface $logger = null, int $maxHostConnections = 6): HttpClientInterface
     {
+        if ($logger === null) {
+            $logger = new NullLogger();
+        }
```

Comment:
```php
$this->logger = $logger ?? new NullLogger();
```
? Consequently with what you did for other clients.

Reply:
logger in not needed as a class property there, it is only in the scope of `create` method

Reply:
Do we really need this? Both constructors of `CurlHttpClient` and `NativeHttpClient` are already handling this case.

Reply:
yes we need, to test classes separately